cmd/clusterctl/client/repository/components_client.go

```diff
@@ -47,37 +47,38 @@ func newComponentsClient(provider config.Provider, repository Repository, config
 	}
 }
 
-func (f *componentsClient) Get(version, targetNamespace, watchingNamespace string) (Components, error) {
+// Get returns the components from a repository without variable replacement
+func (f *componentsClient) Get(options ComponentsOptions) (Components, error) {
```

Comment:
I suggest removing "without variable replacement" from the godoc.

This function doesn't override the `options.SkipVariables=false`. So if `Get(options)` was called with `options.SkipVariables=true` it would I'm assuming replace the variables.